frontend/src/metabase/query_builder/actions/downloading.ts

```diff
@@ -76,92 +135,101 @@ const getDatasetParams = ({
   visualizationSettings,
 }: DownloadQueryResultsOpts): DownloadQueryResultsParams => {
   const cardId = question.id();
-  const isQuestionInStaticEmbedDashboard =
-    dashcardId != null && cardId != null && token != null;
 
   // Formatting is always enabled for Excel
   const format_rows = enableFormatting && type !== "xlsx" ? "true" : "false";
 
-  if (isQuestionInStaticEmbedDashboard) {
-    return {
-      method: "GET",
-      url: `/api/embed/dashboard/${token}/dashcard/${dashcardId}/card/${cardId}/${type}`,
-      params: Urls.getEncodedUrlSearchParams({ ...params, format_rows }),
-    };
+  const { accessedVia, resourceType: resource } = getDownloadedResourceType({
+    dashboardId,
+    dashcardId,
+    uuid,
+    token,
+    question,
+  });
+
+  // Public links use special endpoints that use uuids instead of ids
+  if (accessedVia === "public-link") {
+    if (resource === "dashcard") {
+      return {
+        method: "POST",
+        url: `/api/public/dashboard/${dashboardId}/dashcard/${dashcardId}/card/${cardId}/${type}`,
+        params: new URLSearchParams({ format_rows }),
+        body: {
+          parameters: result?.json_query?.parameters ?? [],
+        },
+      };
+    }
+    if (resource === "question" && uuid) {
+      return {
+        method: "GET",
+        url: Urls.publicQuestion({ uuid, type, includeSiteUrl: false }),
+        params: new URLSearchParams({
+          parameters: JSON.stringify(result?.json_query?.parameters ?? []),
+          format_rows,
+        }),
+      };
+    }
+  }
+
+  // Static embeds use special endpoints that use signed tokens instead of ids
+  if (accessedVia === "static-embed") {
+    if (resource === "dashcard") {
+      return {
+        method: "GET",
+        url: `/api/embed/dashboard/${token}/dashcard/${dashcardId}/card/${cardId}/${type}`,
+        params: Urls.getEncodedUrlSearchParams({ ...params, format_rows }),
+      };
+    }
+
+    if (resource === "question" && token) {
```

Comment:
Yes exactly, without that check it could be undefined, I thought for a moment to make a more complex type to not need that, but it would require a lot of rewrite and I don't think it's worth it

Reply:
That's fair